solax/inverter.py

```diff
@@ -15,6 +16,23 @@ class InverterError(Exception):
 
 InverterResponse = namedtuple("InverterResponse", "data, serial_number, version, type")
 
+_KEY_DATA = "Data"
+_KEY_SERIAL = "SN"
+_KEY_VERSION = "version"
+_KEY_VER = "ver"
+_KEY_TYPE = "type"
+
+
+DataSchema = vol.Schema(
+    {
+        vol.Required(_KEY_DATA): vol.Schema(contains_none_zero_value),
```

Comment:
I'm wondering if it might be easiest to support this on all inverters if we do two passes of the schema validation:
1. in the base inverter, just to check that `'data`' conforms to `contains_none_zero_value`.
2. then the second one where each inverter defines it's own schema

Reply:
Yeah that was the general idea. The presented  Schema basically validates, that a response contains the keys that are required for further processing. Since I moved the creation of the response into the base Inverter class these are all currently possible variations. see `handle_reponse`

Reply:
The updated version or this PR extends the specific schema with required basics.